src/System.Net.Requests/src/System/Net/HttpWebRequest.cs

```diff
@@ -269,11 +280,51 @@ public string Host
         {
             get
             {
-                throw new PlatformNotSupportedException();
+                Uri hostUri = _hostUri ?? Address;
+                return (_hostUri == null || !_hostHasPort) && Address.IsDefaultPort ?
+                    hostUri.Host :
+                    $"{hostUri.Host}:{hostUri.Port}";
```

Comment:
I guess the most performant way without creating a side string would be with StringBuilder?